src/ui/include/EditorNS/jsproxy.h

```diff
@@ -0,0 +1,97 @@
+#ifndef _JSPROXY_H_
+#define _JSPROXY_H_
+
+#include <QObject>
+#include <QVariant>
+#include <QPair>
+
+namespace EditorNS {
+    /**
+     * @brief An Object injectable into the javascript page, that allows
+     *        the javascript code to send messages to an Editor object.
+     *        It also allows the js instance to retrieve message data 
+     *        information.
+    */
+    
+    typedef QPair<int, int> CursorPosition;
+
+    class JsToCppProxy : public QObject
+    {
+        Q_OBJECT
+
+    private:
+        QVariant m_msgData;
+        QVariant m_result;
+
+        // Some editor data we push on the fly
+        QVariant m_cursor;
+        QVariant m_textLength;
+        QVariant m_lineCount;
+        QVariant m_selections;
+        QVariant m_selectionsText;
+        QVariant m_scrollPosition;
+        QVariant m_language;
+        QVariant m_clean;
+        QVariant m_detectedIndent;
+
+    public:
+        JsToCppProxy(QObject *parent) : QObject(parent) { }
+
+        // These are our data retrieval mechanisms.
+        QVariant getResult();
+        QVariant getCursor();
+        QVariant getTextLength();
+        QVariant getLineCount();
+        QVariant getSelections();
+        QVariant getSelectionsText();
+        QVariant getScrollPosition();
+        QVariant getLanguage();
+        QVariant getClean();
+        QVariant getDetectedIndent();
```

Comment:
Was just an example. `getDetectindent` can return a `QPair<bool, int>` for example.

My motivation is to limit passing around QVariants or QVariantMaps as much as possible by doing the conversion to "proper" data types as soon as possible. 

That way none will have to make the mistake I did and assume that DetectIndent's QVariant is a bool value.